src/google/protobuf/util/internal/utility.cc

```diff
@@ -302,7 +302,7 @@ bool IsMap(const google::protobuf::Field& field,
 string DoubleAsString(double value) {
   if (value == std::numeric_limits<double>::infinity()) return "Infinity";
   if (value == -std::numeric_limits<double>::infinity()) return "-Infinity";
-  if (::isnan(value)) return "NaN";
+  if (std::isnan(value)) return "NaN";
```

Comment:
It does not seem to work.  I can see in other places, such as strutil.cc
that IsNaN is locally defined.  Could that be a possibility:

inline bool IsNaN(double value) {
  // NaN is never equal to anything, even itself.
  return value != value;
}

On 29 June 2015 at 22:06, Feng Xiao <notifications@github.com> wrote:

> In src/google/protobuf/util/internal/utility.cc
> <https://github.com/google/protobuf/pull/551#discussion_r33506128>:
>
> > @@ -302,7 +302,7 @@ bool IsMap(const google::protobuf::Field& field,
> >  string DoubleAsString(double value) {
> >    if (value == std::numeric_limits<double>::infinity()) return "Infinity";
> >    if (value == -std::numeric_limits<double>::infinity()) return "-Infinity";
> > -  if (::isnan(value)) return "NaN";
> > +  if (std::isnan(value)) return "NaN";
>
> Could you change it to "MathLimits::IsNan(value)"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/protobuf/pull/551/files#r33506128>.
>